react/utils/isAdmin.ts

```diff
@@ -0,0 +1,28 @@
+import { canUseDOM } from 'exenv'
+
+/**
+ * This function checks if anything is running in the Admin environment,
+ * without relying on Render's own features.
+ */
+export function isAdmin() {
+  if (canUseDOM) {
+    const { host } = window.location
+
+    const domain = host.match(/(?<=\.)([^.]+\.[^.]+$)/)?.[0]
+
+    if (domain !== 'myvtex.com') {
+      return false
+    }
+
+    const adminPathnames = [
+      '/admin',
+      '/_v/segment/admin-login/v1/login', // Let's consider the Admin Login page as an Admin App even though it's technically not.
+    ]
+
+    const { pathname } = window.location
+
+    return adminPathnames.some((path) => pathname.startsWith(path))
+  }
+
+  return false
```

Comment:
```suggestion
  if (!canUseDOM) return false
  
  const { host } = window.location
  const domain = host.match(/(?<=\.)([^.]+\.[^.]+$)/)?.[0]

  if (domain !== 'myvtex.com') {
    return false
  }

  const adminPathnames = [
    '/admin',
    '/_v/segment/admin-login/v1/login', // Let's consider the Admin Login page as an Admin App even though it's technically not.
  ]

  const { pathname } = window.location

  return adminPathnames.some((path) => pathname.startsWith(path))
```

In find the early return is often better to read. Just a suggestion, apply if you think it looks better.

Reply:
Much better indeed @matheusps! Thanks. I'm applying it.